app/main/main.js

```diff
@@ -0,0 +1,35 @@
+angular.module('timeshare', [
+  "firebase", 'ui.router'
+])
+.controller('MainController', function($scope, $firebaseObject){
+  
+  $scope.data = $firebaseObject(ref);
+
+});
+
+var ref = new Firebase("https://timeshare.firebaseio.com");
+var users = ref.child('users');
+ 
+function fbLogin() {
+   ref.authWithOAuthPopup("facebook", function(error, authData) {
+     if (error) {
+       console.log("Login Failed!", error);
+     } else {
+       console.log("Authenticated successfully with payload:", authData);
+       users.set({
+         name: authData.facebook.displayName,
+         id: authData.facebook.id
+       });
+     }
+   });
+ }
+
+
+
+ ref.on('child_added', function(snapshot, prevChildKey) {
+   var newUser = snapshot.val()
+   console.log('a new user named ' + newUser.name + ' was created');
+ });
```

Comment:
@gdomorski are we still using this code? if so it needs to be in the controller as well